### PR TITLE
add --no-sandbox to smoke test

### PR DESCRIPTION
**Pull request by flotwig**
- Closes 

### Pre-merge Tasks

- [x] Have tests been added/updated for the changes in this PR?
- [ ] Has a PR to [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation) been submitted to document any user-facing changes? 
- [ ] Have the [type definitions](cli/types/index.d.ts) been updated with any user-facing API changes?
- [ ] Has the [cypress.schema.json](cli/schema/cypress.schema.json) been updated with any new configuration options?
- [ ] Has the original issue been tagged with a release in ZenHub?

### CODE REVIEW

File: cli/lib/tasks/verify.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ const runSmokeTest = (binaryDir, options) => {
    */
   const spawn = (linuxWithDisplayEnv) => {
     const random = _.random(0, 1000)
-    const args = ['--smoke-test', `--ping=${random}`]
+    const args = ['--no-sandbox', '--smoke-test', `--ping=${random}`]
```
Comment (bahmutov) [inline cli/lib/tasks/verify.js]: I would add comments above why `--no-sandbox` is necessary

Comment (bahmutov) [reply cli/lib/tasks/verify.js]: also, ughh, I hate that it generated floating-point number
```
Error: Command failed: /root/cypress/build/linux/Cypress/Cypress --smoke-test --ping=0.6115562798944125
```
How is it possible if https://lodash.com/docs/4.17.15#random says it should be an integer?!

Comment (flotwig) [reply cli/lib/tasks/verify.js]: updated the code with a comment + a conditional

File: cli/lib/tasks/verify.js [hunk 2]
```diff
@@ -79,6 +79,11 @@ const runSmokeTest = (binaryDir, options) => {
   const spawn = (linuxWithDisplayEnv) => {
     const random = _.random(0, 1000)
     const args = ['--smoke-test', `--ping=${random}`]
+    
+    if (process.platform !== 'win32' && process.geteuid() === 0) {
```
Comment (bahmutov) [inline cli/lib/tasks/verify.js]: I would prefer positive conditions, and more readable code, for example
```js
const isLinuxLike = () => process.platform() !== 'win32'

const isRootUser = () => process.geteuid() === 0

if (isLinuxLike() && isRootUser()) {
  // sandbox code
}
```

Comment (bahmutov) [reply cli/lib/tasks/verify.js]: still good, no worries

File: cli/test/lib/tasks/verify_spec.js [hunk 3]
```diff
@@ -53,6 +53,7 @@ context('lib/tasks/verify', () => {
     sinon.stub(xvfb, 'stop').resolves()
     sinon.stub(xvfb, 'isNeeded').returns(false)
     sinon.stub(Promise.prototype, 'delay').resolves()
+    sinon.stub(process, 'geteuid').returns(1000)
```
Comment (bahmutov) [inline cli/test/lib/tasks/verify_spec.js]: well, if we have a test for "regular" user, we probably should have a test for root user?

File: cli/lib/tasks/verify.js [hunk 4]
```diff
@@ -80,6 +80,11 @@ const runSmokeTest = (binaryDir, options) => {
     const random = _.random(0, 1000)
     const args = ['--smoke-test', `--ping=${random}`]
 
+    if (process.platform !== 'win32' && process.geteuid() === 0) {
+      // electron requires --no-sandbox to run as root
+      args.unshift('--no-sandbox')
```
Comment (bahmutov) [inline cli/lib/tasks/verify.js]: why `unshift` and not just `args.push`?

Comment (flotwig) [reply cli/lib/tasks/verify.js]: no reason, either should work fine

File: cli/test/lib/tasks/verify_spec.js (hunk 3 above)
Comment (flotwig) [reply cli/test/lib/tasks/verify_spec.js]: added a test for euid = 0

**Pull request merged**